src/Promitor.Tests.Unit/Azure/AzureAuthenticationFactoryUnitTests.cs

```diff
@@ -210,6 +210,52 @@ public void GetConfiguredAzureAuthentication_ServicePrincipleIsValidWithLegacyAn
             Assert.Equal(expectedSecret, authenticationInfo.Secret);
         }
 
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData(null)]
+        public void GetConfiguredAzureAuthentication_ServicePrincipleWithInvalidSecretFilePath_Fails(string secretFilePath)
+        {
+            // Arrange
+            var expectedIdentityId = Guid.NewGuid().ToString();
+            var expectedAuthenticationMode = AuthenticationMode.ServicePrincipal;
+            var expectedSecretFileName = Guid.NewGuid().ToString();
+            var inMemoryConfiguration = new Dictionary<string, string>
+            {
+                {ConfigurationKeys.Authentication.Mode, expectedAuthenticationMode.ToString()},
+                {EnvironmentVariables.Authentication.ApplicationId, expectedIdentityId},                
+                {ConfigurationKeys.Authentication.SecretFilePath, secretFilePath},
+                {ConfigurationKeys.Authentication.SecretFileName, expectedSecretFileName}
+            };
+            var config = CreateConfiguration(inMemoryConfiguration);
+
+            // Act & Assert
+            Assert.Throws<AuthenticationException>(() => AzureAuthenticationFactory.GetConfiguredAzureAuthentication(config));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData(null)]
+        public void GetConfiguredAzureAuthentication_ServicePrincipleWithInvalidSecretFileName_Fails(string secretFileName)
```

Comment:
Can we add a positive test to ensure that it works as well please?

Reply:
I wanted to but since it checks the existance of a file on the file system, how do you suggest to make this test?